org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/dom/HierarchicalASTVisitor.java

```diff
@@ -156,22 +156,14 @@ public void endVisit(TypeDeclaration node) {
 
 	//---- Begin AbstractUnnamedTypeDeclaration Hierarchy ---------------------------
 
-	public boolean visit(AbstractUnnamedTypeDeclaration node) {
-		return visit((BodyDeclaration)node);
-	}
-
-	public void endVisit(AbstractUnnamedTypeDeclaration node) {
-		endVisit((BodyDeclaration)node);
-	}
-
 	@Override
 	public boolean visit(ImplicitTypeDeclaration node) {
-		return visit((AbstractUnnamedTypeDeclaration)node);
+		return visit((AbstractTypeDeclaration)node);
```

Comment:
I've removed these two methods in https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/1420.
Should I add them back like here?

Reply:
I've checked in the IDE, yes, I have to add them. I will update my PR

Reply:
Okay.